hail/python/hail/vds/sample_qc.py

```diff
@@ -0,0 +1,310 @@
+    dp_bins=sequenceof(int),
+)
+def rmt_sample_qc(
+    *,
+    locus: 'LocusExpression',
+    end: 'NumericExpression',
+    gq: 'NumericExpression',
+    dp: Optional['Expression'] = None,
+    gq_bins: 'Sequence[int]' = (0, 20, 60),
+    dp_bins: 'Sequence[int]' = (0, 1, 10, 20, 30),
+) -> 'StructExpression':
+    ref_dp_expr = {}
+    if dp is not None:
+        ref_dp_expr['ref_bases_over_dp_threshold'] = hl.tuple(
+            hl.agg.filter(dp >= x, hl.agg.sum(1 + end - locus.position)) for x in dp_bins
+        )
+    return hl.struct(
+        bases_over_gq_threshold=hl.tuple(hl.agg.filter(gq >= x, hl.agg.sum(1 + end - locus.position)) for x in gq_bins),
+        **ref_dp_expr,
+    )
+
+
+def combine_sample_qc(
+    rmt_sample_qc: Expression,
+    vmt_sample_qc: Expression,
+) -> Expression:
+    assert 'gq_dp_exprs' in vmt_sample_qc
+    assert 'bases_over_gq_threshold' in rmt_sample_qc
+
+    joined_dp_expr = {}
+    if 'bases_over_dp_threshold' in vmt_sample_qc:
+        joined_dp_expr['bases_over_dp_threshold'] = hl.tuple(
+            x + y for x, y in zip(vmt_sample_qc.gq_dp_exprs.dp, rmt_sample_qc.bases_over_dp_threshold)
+        )
+
+    return hl.struct(
+        bases_over_gq_threshold=hl.tuple(
+            x + y for x, y in zip(vmt_sample_qc.gq_dp_exprs.gq, rmt_sample_qc.bases_over_gq_threshold)
+        ),
+        **joined_dp_expr,
+    )
+
+
+@typecheck(vds=VariantDataset, gq_bins=sequenceof(int), dp_bins=sequenceof(int), dp_field=nullable(str))
+def sample_qc(
+    vds: 'VariantDataset',
+    *,
+    gq_bins: 'Sequence[int]' = (0, 20, 60),
+    dp_bins: 'Sequence[int]' = (0, 1, 10, 20, 30),
+    dp_field=None,
+) -> 'Table':
+    """Compute sample quality metrics about a :class:`.VariantDataset`.
+
+    If the `dp_field` parameter is not specified, the ``DP`` is used for depth
+    if present. If no ``DP`` field is present, the ``MIN_DP`` field is used. If no ``DP``
+    or ``MIN_DP`` field is present, no depth statistics will be calculated.
+
+    Parameters
+    ----------
+    vds : :class:`.VariantDataset`
+        Dataset in VariantDataset representation.
+    name : :obj:`str`
+        Name for resulting field.
+    gq_bins : :class:`tuple` of :obj:`int`
+        Tuple containing cutoffs for genotype quality (GQ) scores.
+    dp_bins : :class:`tuple` of :obj:`int`
+        Tuple containing cutoffs for depth (DP) scores.
+    dp_field : :obj:`str`
+        Name of depth field. If not supplied, DP or MIN_DP will be used, in that order.
+
+    Returns
+    -------
+    :class:`.Table`
+        Hail Table of results, keyed by sample.
+    """
+
+    require_first_key_field_locus(vds.reference_data, 'sample_qc')
+    require_first_key_field_locus(vds.variant_data, 'sample_qc')
+
+    if 'DP' in vds.reference_data.entry:
+        ref_dp_field_to_use = 'DP'
+    elif 'MIN_DP' in vds.reference_data.entry:
+        ref_dp_field_to_use = 'MIN_DP'
+    else:
+        ref_dp_field_to_use = dp_field
```

Comment:
I know this was the old behavior, but from the docstring I would expect `dp_field` to always be used if present. Should either reorder the branches here, or reword the docs.